Extendable static files

vilppuvuorinen opened the pull request:
`JettyResourceHandler` implementation does not fit my use case. I would like to be able to supply my own implementation for `JavalinServlet`.  

It would probably be a _cleaner_ solution if the entire `JettyResourceHandler` was moved behind an interface and adding a setter to the `Javalin` object. In my opinion that would be just overkill.

tipsy commented:
> I would like to be able to supply my own implementation for JavalinServlet.

Do you already have an implementation? It would be great it we could move away from using Jetty for static resources altogether.

tipsy commented:
Sure, go ahead and make a pull request.

vilppuvuorinen commented:
I removed the `open` declarations from `JettyResourceHandler` and replaced it with a `ResourceHandler` interface. Now it should be possible to extend `Javalin` class and provide custom implementation. The change did not seem to affect tests and I did not feel like it would warrant it's own tests as taking advantage of this functionality requires extending `Javalin`.

```java
public class CustomJavalin extends Javalin {
    @NotNull
    @Override
    public JavalinServlet createServlet() {
        return new JavalinServlet(
            this,
            pathMatcher,
            exceptionMapper,
            errorMapper,
            debugLogging,
            requestLogger,
            dynamicGzipEnabled,
            autogeneratedEtagsEnabled,
            defaultContentType,
            maxRequestCacheBodySize,
            prefer405over404,
            singlePageHandler,
            new CustomResourceHandler(staticFileConfig, jettyServer, ignoreTrailingSlashes)
        );
    }
}
```

I'm not 100% sure about naming tho.

tipsy commented:
Thanks @vilppuvuorinen !

tipsy commented:
This has been released as part of `2.5.0` now: https://javalin.io/news/2019/01/04/javalin-2.5.0-released.html
